File: 序号/377. 组合总和 Ⅳ.java
class Solution {
    public int combinationSum4(int[] nums, int t) {
        int[] f = new int[t + 1];
        f[0] = 1;
        for (int j = 1; j <= t; j++) {
            for (int u : nums) {
                if (j >= u) f[j] += f[j - u];
            }
        }
        return f[t];
    }


}

File: 序号/377. 组合总和 Ⅳ.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        @cache
        def dfs(t):
            if t<0:return 0
            if t==0:return 1
            ans=0
            for x in nums:
                ans+=dfs(t-x)
            return ans
        return dfs(target)
